### PR TITLE
More frontend stuff

### PR DESCRIPTION
**Pull request by ConnorCimowsky**

### CODE REVIEW

File: src-frontend/app/controllers/add-element-modal.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+        eid: this.get('eid'),
+        displayIndex: this.get('displayIndex'),
+        elementType: this.get('elementType'),
+        concept: this.get('concept'),
+        question: this.get('question'),
+        answer: this.get('answer')
+      });
+
+      var choices = this.get('choices');
+      if (choices !== "") {
```
Comment (Trinovantes) [inline src-frontend/app/controllers/add-element-modal.js]: so we're using a delimited string for the list of choices?

Comment (ConnorCimowsky) [inline src-frontend/app/controllers/add-element-modal.js]: For the demo, yeah :/ I'll add a TODO

Comment (Trinovantes) [inline src-frontend/app/controllers/add-element-modal.js]: can use `!!choices` here

File: src-frontend/app/controllers/procedure.js [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
+    addElement: function(page, selectedIndex) {
+      var targetIndex = selectedIndex + 1;
+      var elements = page.get('elements');
+      var elementsToBePushed = elements.filter(function(element) {
+        return element.get('displayIndex') >= targetIndex;
+      });
+
+      elementsToBePushed.forEach(function(element) {
+        element.incrementProperty('displayIndex');
+        element.save();
```
Comment (Trinovantes) [inline src-frontend/app/controllers/procedure.js]: I guess we can't bulk save here? :(

File: src-frontend/app/serializers/page.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import DS from 'ember-data';
+
+export default DS.RESTSerializer.extend({
+  serialize: function(snapshot, options) {
+    var json = this._super(snapshot, options);
+
+    if (json.elements === undefined) {
```
Comment (Trinovantes) [inline src-frontend/app/serializers/page.js]: I think `!json.elements` is better here

**Comment by Trinovantes** [comment]
+1

File: src-frontend/app/controllers/procedure.js (hunk 2 above)
Comment (ConnorCimowsky) [inline src-frontend/app/controllers/procedure.js]: I was told that Ember batches them up in the current run loop. But I'm still getting weird behaviour if I cascade a bunch of elements. I'll add a TODO

**Pull request merged**